.github/workflows/release.yml

```
@@ -34,23 +35,60 @@ jobs:
           name: XenAPI
           path: scripts/examples/python/dist/
 
+  build-sdks:
+    name: Build and upload SDK artifacts
+    uses: ./.github/workflows/generate-and-build-sdks.yml
+    with:
+      xapi_version: ${{ github.ref_name }}
 
   release:
+    name: "Create and package release"
     runs-on: ubuntu-latest
-    needs: build-python
+    needs: [build-python, build-sdks]
     steps:
-      - name: Retrieve python distribution artifacts
+      - name: Retrieve Python SDK distribution artifacts
         uses: actions/download-artifact@v3
         with:
           name: XenAPI
           path: dist/
 
+      - name: Retrieve C# SDK distribution artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: SDK_Binaries_CSharp
+          path: dist/
+
+      - name: Retrieve PowerShell 5.x SDK distribution artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: XenServerPowerShell_NET45
+          path: sdk_powershell_5x/
+
+      - name: Retrieve PowerShell 7.x SDK distribution artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: XenServerPowerShell_NET8
+          path: sdk_powershell_7x/
+
+      - name: Zip PowerShell 5.x SDK artifacts for deployment
+        shell: bash
+        run: zip PowerShell-SDK-5.x-prerelease-unsigned.zip ./sdk_powershell_5x -r
+      
+      - name: Zip PowerShell 7.x SDK artifacts for deployment
+        shell: bash
+        run: zip PowerShell-SDK-7.x-prerelease-unsigned.zip ./sdk_powershell_7x -r
+
       - name: Create release ${{ github.ref_name }}
-        run: gh release create ${{ github.ref_name }} --repo ${{ github.repository }} --generate-notes dist/*
+        shell: bash
+        run: |
+          gh release create ${{ github.ref_name }} --repo ${{ github.repository }} --generate-notes dist/*
+          gh release upload ${{ github.ref_name }} --repo ${{ github.repository }} PowerShell-SDK-5.x-prerelease-unsigned.zip
+          gh release upload ${{ github.ref_name }} --repo ${{ github.repository }} PowerShell-SDK-7.x-prerelease-unsigned.zip
```

There's quite a bit of repetition here, this would work as well:
```suggestion
gh release create ${{ github.ref_name }} --repo ${{ github.repository }} --generate-notes dist/* PowerShell-SDK-5.x-prerelease-unsigned.zip PowerShell-SDK-7.x-prerelease-unsigned.zip
```

If you want to have an artifact per line, maybe using `/` could work?

---

Well spotted. I had assumed (but not checked) that I couldn't list them like that!